### PR TITLE
fix: add missing await in src/gridfs

### PR DESCRIPTION
**Pull request by c0per**
I found the code for gridfs has some promises without await. I made a small fix to add those awaits.

### CODE REVIEW

**Comment by erfanium** [comment]
@lucsoft

**Comment by lucsoft** [comment]
You don't need Return await it's harming performance and needs it's pressures the GC

**Comment by lucsoft** [comment]
@c0per did you had problems with the missing awaits?
VS Code show still show you that you need to await even though its not a "async function"

File: src/gridfs/bucket.ts [hunk 1]
```diff
@@ -66,12 +60,12 @@ export class GridFSBucket {
...
    * Returns a Stream to which the application will write the contents.
    */
-  openUploadStreamWithId(
+  async openUploadStreamWithId(
     id: FileId,
     filename: string,
     options?: GridFSUploadOptions,
   ) {
-    if (!this.#checkedIndexes) this.#checkIndexes();
+    if (!this.#checkedIndexes) await this.#checkIndexes();
```
Comment (lucsoft) [inline src/gridfs/bucket.ts:68]: This only should be the issue. And the functions referencing to openUploadStreamWithId

File: src/gridfs/indexes.ts [hunk 2]
```diff
@@ -31,18 +35,24 @@ export async function checkIndexes(
...
+      projection: { _id: 1 },
+    },
+  ));
 
   if (filesCollectionIsEmpty || chunksCollectionIsEmpty) {
     // At least one collection is empty so we create indexes
-    createFileIndex(filesCollection);
-    createChunksIndex(chunksCollection);
+    await createFileIndex(filesCollection);
+    await createChunksIndex(chunksCollection);
```
Comment (lucsoft) [inline src/gridfs/indexes.ts:55]: oh yeah and both functions require await but they don't require to be async function

**Review by lucsoft — CHANGES_REQUESTED** [review]
clean up the return await and then it looks good 🥳

File: src/gridfs/bucket.ts (hunk 1 above)
Comment (c0per) [reply src/gridfs/bucket.ts:68]: Yes. I encountered this bug.

File: src/gridfs/indexes.ts (hunk 2 above)
Comment (c0per) [reply src/gridfs/indexes.ts:55]: I fixed this in the next commit.

**Review by lucsoft — CHANGES_REQUESTED** [review]

File: src/gridfs/bucket.ts [hunk 3]
```diff
@@ -176,10 +171,10 @@ export class GridFSBucket {
     await this.#chunksCollection.drop();
   }
 
-  #checkIndexes = () =>
-    checkIndexes(
+  #checkIndexes = async () =>
+    await checkIndexes(
```
[the remark this answers is not in the document]
Comment (c0per) [reply src/gridfs/bucket.ts]: Ah, yes. I fixed that. 😅

**Review by lucsoft — APPROVED** [review]
LGTM :shipit:

**Pull request merged**